`husky` seems to be breaking `precommit`

If you

1. Install server environment and setup precommit (`pip install -e server[dev]`, `pre-commit install` etc)
2. Install client environment (`pnpm install`)

And then try to commit something in any part of the repo, you'll notice that the server precommit hooks no longer fire, even if you have, say, intentionally introduced trailing whitespace or something of that nature:

```
[ feat/475-deploy-frontend ⚙ .venv] code/metakb/server % git commit -m 'fix things?'

> metakb@ lint-staged /Users/jss009/code/metakb
> lint-staged

✔ Backed up original state in git stash (5e9c2424)
✔ Running tasks for staged files...
✔ Applying modifications from tasks...
✔ Cleaning up temporary files...
[feat/475-deploy-frontend 2e615aad] fix things?
 3 files changed, 19 insertions(+), 11 deletions(-)
```

And in fact, trying to reinstall precommit hooks will fail:

```
[ testing-precommit ⚙ .venv] ~/code/metakb % pre-commit install
[ERROR] Cowardly refusing to install hooks with `core.hooksPath` set.
hint: `git config --unset-all core.hooksPath`
```

Running `pre-commit run` still works, but it appears that `husky` overwrites the git repo hooks in a way that removes precommit. This makes sense, they're sort of redundant. But it's unfortunate because it means we lose a lot of precommit hooks that we like for code quality, like trailing whitespace, line ending standardization, etc.

A few ways forward

1. Add the precommit hooks into the husky config. I imagine this is possible with some doing, the docs seem very light on details but I'm sure it's possible
2. Try to add the husky behavior in precommit. I think I know how to do this.
3. Try something like [prek](https://github.com/j178/prek)

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -63,7 +63,7 @@ MetaKB uses automated tooling to maintain consistent code quality across the pro
 - To manually check or fix:
 
   ```bash
-  ruff check server
+  ruff check --fix server
   ruff format server
   ```
 

--- a/package.json
+++ b/package.json
@@ -4,7 +4,7 @@
   "scripts": {
     "dev": "turbo run dev --parallel",
     "prepare": "husky",
-    "lint-staged": "lint-staged",
+    "lint-staged": "lint-staged --fail-on-changes --continue-on-error --no-revert",
     "format": "pnpm --filter client format && ruff check server --fix && ruff format server"
   },
   "devDependencies": {

--- a/package.json
+++ b/package.json
@@ -15,8 +15,11 @@
   },
   "lint-staged": {
     "server/**/*.py": [
-      "ruff check --fix",
-      "ruff format"
+      "ruff format",
+      "ruff check --fix --exit-non-zero-on-fix"
+    ],
+    "*": [
+      "pre-commit run check-added-large-files --files"
     ]
   },
   "dependencies": {

--- a/package.json
+++ b/package.json
@@ -19,7 +19,13 @@
       "ruff check --fix --exit-non-zero-on-fix"
     ],
     "*": [
-      "pre-commit run check-added-large-files --files"
+      "pre-commit run check-added-large-files --files",
+      "pre-commit run detect-private-key --files",
+      "pre-commit run trailing-whitespace --files",
+      "pre-commit run end-of-file-fixer --files",
+      "pre-commit run check-merge-conflict --files",
+      "pre-commit run detect-aws-credentials --allow-missing-credentials --files",
+      "pre-commit run mixed-line-ending --fix=lf --files"
     ]
   },
   "dependencies": {

--- a/package.json
+++ b/package.json
@@ -24,8 +24,8 @@
       "pre-commit run trailing-whitespace --files",
       "pre-commit run end-of-file-fixer --files",
       "pre-commit run check-merge-conflict --files",
-      "pre-commit run detect-aws-credentials --allow-missing-credentials --files",
-      "pre-commit run mixed-line-ending --fix=lf --files"
+      "pre-commit run detect-aws-credentials --files",
+      "pre-commit run mixed-line-ending --files"
     ]
   },
   "dependencies": {

